### PR TITLE
Removed `Bert` interdependency in `tokenization_electra.py`

### PR DESCRIPTION
**Pull request by OtherHorizon**
# What does this PR do?

Related to #19303 
Removes `bert` dependency from `tokenization_electra.py`

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
 @sgugger

### CODE REVIEW

File: src/transformers/models/electra/tokenization_electra.py [hunk 1]
```diff
@@ -58,18 +63,459 @@
...
+                    logger.warning(
+                        f"Saving vocabulary to {vocab_file}: vocabulary indices are not consecutive."
+                        " Please check that the vocabulary is not corrupted!"
+                    )
+                    index = token_index
+                writer.write(token + "\n")
+                index += 1
+        return (vocab_file,)
+
+class BasicTokenizer(object):
```
Comment (sgugger) [inline src/transformers/models/electra/tokenization_electra.py:318]: There should be a copied from here to make sure this object stays in sync with the same in `tokenization_bert`.

**Review by sgugger — COMMENTED** [review]
Thanks a lot for working on this one! It's missing a few copied froms, I left suggestions.
Once you're done, be sure to run `make style` on your branch (for code-formatting) and we should be good to merge!

File: src/transformers/models/electra/tokenization_electra.py [hunk 2]
```diff
@@ -58,18 +63,459 @@
...
+            if cp == 0 or cp == 0xFFFD or _is_control(char):
+                continue
+            if _is_whitespace(char):
+                output.append(" ")
+            else:
+                output.append(char)
+        return "".join(output)
+
+
+class WordpieceTokenizer(object):
```
[the remark this answers is not in the document]
Comment (OtherHorizon) [reply src/transformers/models/electra/tokenization_electra.py:468]: should it be like `# Copied from transformers.models.bert.tokenization_bert.WordpieceTokenizer`

**Comment by OtherHorizon** [comment]
> Thanks a lot for working on this one! It's missing a few copied froms, I left suggestions. Once you're done, be sure to run `make style` on your branch (for code-formatting) and we should be good to merge!

`make style` changes a lot of files.

File: src/transformers/models/electra/tokenization_electra.py [hunk 3]
```diff
@@ -58,18 +63,461 @@
...
+            if cp == 0 or cp == 0xFFFD or _is_control(char):
+                continue
+            if _is_whitespace(char):
+                output.append(" ")
+            else:
+                output.append(char)
+        return "".join(output)
+
+
+# Copied from transformers.models.bert.tokenization_bert.WordpieceTokenizer with Bert->Electra
```
Comment (sgugger) [inline src/transformers/models/electra/tokenization_electra.py:468]:
```suggestion
# Copied from transformers.models.bert.tokenization_bert.WordpieceTokenizer
```

File: src/transformers/models/electra/tokenization_electra.py [hunk 4]
```diff
@@ -58,18 +63,461 @@
...
+                if index != token_index:
+                    logger.warning(
+                        f"Saving vocabulary to {vocab_file}: vocabulary indices are not consecutive."
+                        " Please check that the vocabulary is not corrupted!"
+                    )
+                    index = token_index
+                writer.write(token + "\n")
+                index += 1
+        return (vocab_file,)
+# Copied from transformers.models.bert.tokenization_bert.BasicTokenizer with Bert->Electra
```
Comment (sgugger) [inline src/transformers/models/electra/tokenization_electra.py:318]:
```suggestion
# Copied from transformers.models.bert.tokenization_bert.BasicTokenizer
```

**Review by sgugger — COMMENTED** [review]
Just a last couple of nits and then it's up to the quality checks to be green before we can emrge this :-)

- run `make quality` to make sure the formatting test passes (normally all errors should be fixed by `make style` but sometime you have to manually do something)
- run `make repo-consistency` to make all the other quality scripts for the repo pass (normally running `make fix-copies` should fix all errors)

File: src/transformers/models/electra/tokenization_electra.py [hunk 5]
```diff
@@ -13,7 +13,13 @@
...
 # limitations under the License.
 
-from ..bert.tokenization_bert import BertTokenizer
+import collections
+import os
+import unicodedata
+from typing import List, Optional, Tuple
+
+from ...tokenization_utils import PreTrainedTokenizer, _is_control, _is_punctuation, _is_whitespace
+from ...utils import logging
```
Comment (sgugger) [inline src/transformers/models/electra/tokenization_electra.py:22]: You need to copy the line setting the logger for this module too :-)

File: src/transformers/models/electra/tokenization_electra.py [hunk 6]
```diff
@@ -58,18 +64,466 @@
     "google/electra-large-discriminator": {"do_lower_case": True},
 }
 
+# Copied from transformers.models.bert.tokenization_bert.load_vocab
```
Comment (sgugger) [inline src/transformers/models/electra/tokenization_electra.py:67]:
```suggestion

# Copied from transformers.models.bert.tokenization_bert.load_vocab
```
flake8 wants a second blank line ehre

**Review by sgugger — COMMENTED** [review]
Almost done, just code-style fixes to add and we should be good to merge!

**Review by sgugger — APPROVED** [review]
Yay, all clean! Thanks a lot for your work on this!

**Pull request merged**